src/torchcodec/decoders/_core/ops.py

```diff
@@ -15,30 +18,61 @@
     _get_extension_path,
 )
 
+_pybind_ops: Optional[ModuleType] = None
 
-def load_torchcodec_extension():
-    # Successively try to load libtorchcodec7.so, libtorchcodec6.so,
-    # libtorchcodec5.so, and libtorchcodec4.so. Each of these correspond to an
+
+def load_torchcodec_shared_libraries():
+    # Successively try to load libtorchcodec_*7.so, libtorchcodec_*6.so,
+    # libtorchcodec_*5.so, and libtorchcodec_*4.so. Each of these correspond to an
     # ffmpeg major version. This should cover all potential ffmpeg versions
     # installed on the user's machine.
     #
     # On fbcode, _get_extension_path() is overridden and directly points to the
     # correct .so file, so this for-loop succeeds on the first iteration.
+    #
+    # Note that we use two different methods for loading shared libraries:
+    #
+    #   1. torch.ops.load_library(): For PyTorch custom ops. Loading libraries
+    #      through PyTorch registers the custom ops with PyTorch's runtime and
+    #      the ops can be accessed through torch.ops after loading.
+    #
+    #   2. importlib: For pybind11 modules. We load them dynamically, rather
+    #      than using a plain import statement. A plain import statement only
+    #      works when the module name and file name match exactly, and the
+    #      shared library file is in the import path. Our shared libraries do
+    #      not meet those conditions.
```

Comment:
Nit: I think we should remove this part from the sentence:

> and the shared library file is in the import path

Because the so/dylib files we produce are definitely in the "import path" (they're right next to other files we happily import). My point being that the above is *not* one of the conditions we fail to meet.